tests/test_linear4bit.py

```diff
@@ -146,3 +147,18 @@ def test_linear_serialization(quant_type, compress_statistics, bias, quant_stora
         target_compression = 0.143 if original_dtype == torch.float32 else 0.29  # these numbers get lower as weight shape increases
         ratio_error_msg = f"quantized_size {size_4:,} is larger on disk than {target_compression:.2%} of original size {size_orig:,}"
         assert size_ratio < target_compression, ratio_error_msg
+           
+def test_copy_param():
+    tensor = torch.tensor([1.,2.,3.,4.])
+    param = bnb.nn.Params4bit(data = tensor, requires_grad=False).cuda(0)
```

Comment:
Is `.cuda(0)` required for these tests?

Reply:
We need to move the param to `cuda` to that the quantization happens and the `quant_state` is initialized. The main pain point was that the `quant_state` was not deepcopied and we get `None` instead.

Reply:
Gotcha, thanks for the clarification.